### ISSUE TITLE
Infinity loss when training EDSR-SuperRes model with float16

### ISSUE THREAD

**Issue by Golbstein** [keras-team/keras#11832]
Hi, I'm training my implementation of [EDSR](https://arxiv.org/abs/1707.02921) on V100 but unfortunately I can't train the model when I set my FP to 16. 

Code: [https://github.com/Golbstein/EDSR-Keras](url)
I tried to change the RELU to Tanh and also tweaked the activity\kernel regularization without success. 

The model does converge when training with float32 but it is x2 time slower. 
When I train with float16, the loss goes down for about 10 mini-batches and then it suddenly start go up until I get 'nan' and 'inf' values..

any advice will be greatly appreciated

**Comment by Golbstein** [comment]
Solved: need to set - 
`K.set_epsilon(1e-4)`
